### PR TITLE
snipsync hello mtls

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?

From this [SDK plan](https://github.com/temporalio/sdk-features/issues/89)
Awaits on the following merges:

- [ ] [Go](https://github.com/temporalio/samples-go/pull/214)
- [ ] [Java](https://github.com/temporalio/samples-java/pull/336)
- [ ] [Python](https://github.com/temporalio/samples-python/pull/10)
- [ ] [TypeScript](https://github.com/temporalio/samples-typescript/pull/187)

## Notes to reviewers

Will open Draft PR once all libraries are mreged.

### CODE REVIEW

**Review by flossypurse — COMMENTED** [review]
The "cleanliness" of these samples - is not really up to my standard for the app dev guide.

I think we should either add them directly to the docs files for now, or rework these samples to be cleaner.

**Comment by bergundy** [comment]
> 

I totally agree.

Patrick, can you please either fix the samples or let us know which changes should be made?

File: docs/application-development/foundations.md [hunk 1]
```diff
@@ -648,34 +637,10 @@ To set the mTLS configuration in Java, provide the certificate and private key i
...
-                .setSslContext(SimpleSslContextBuilder.forPKCS8(clientCert, clientKey).build())
-                .setTarget(targetEndpoint)
-                .build());
-
-```
+<!--SNIPSTART java-mtls-configuration-->
+<!--SNIPEND-->
 
-For more information, see [Sample](https://github.com/temporalio/samples-java/blob/main/src/main/java/io/temporal/samples/ssl/SslEnabledWorker.java).
+For more information, see the [SSL sample](https://github.com/temporalio/samples-java/blob/main/src/main/java/io/temporal/samples/ssl/SslEnabledWorker.java).
```
Comment (bergundy) [inline docs/application-development/foundations.md:643]: Do we need this comment? The source should be linked when using snipsync.

File: docs/application-development/foundations.md [hunk 2]
```diff
@@ -687,18 +652,8 @@ Content is not available
...
-  tls: {
-    clientCertPair: {
-      crt: clientCertPath,
-      key: clientKeyPath,
-    },
-  },
-});
-const client = new WorkflowClient({connection, namespace: "foo.bar"});
-```
+<!--SNIPSTART typescript-mtls-configuration-->
```
Comment (bergundy) [inline docs/application-development/foundations.md:655]: TypeScript docs should mention both client `Connection` and `NativeConnection`

**Pull request closed without merging**